libraries/SRV_Channel/SRV_Channel.h

```diff
@@ -317,6 +318,9 @@ class SRV_Channels {
     // set output value for a specific function channel as a pwm value
     static void set_output_pwm_chan(uint8_t chan, uint16_t value);
 
+    // set output value for a specific function channel as a pwm value for a number of loops
+    static void set_output_pwm_chan_counter(uint8_t chan, uint16_t value, uint16_t counter);
```

Comment:
change counter to timeout_ms, and internally calculate counter using scheduler loop rate

Reply:
milliseconds or seconds I'd say (either is fine).  if you wanted no-timeout I'd personally be OK with the argument being set to zero to indicate this.

Reply:
Seconds would too long for say system identification or fast moving aircraft. Milliseconds at 8 bit would be ridiculously short time. Zero is the current indication of no override. Changing to a signed int would only reduced max possible times even more and half of the memory space used would be nonsensical.

Reply:
I was thinking of a floating point argument for seconds so it wouldn't limit the minimum size.  In any case we've probably discussed all this enough..

Reply:
No it needs to be discussed. This change goes to everyone and it needs to be correct.

The minimum scripting call back time is 1 ms. 400 Hz has a period of 2.5 ms. Until ArduPilot gets above the 1000 Hz mark it doesn't help to allow timeouts less than 1 ms.